internal/cli/push.go

```diff
@@ -83,8 +90,18 @@ var CmdPush = &cobra.Command{
 			return errors.Wrap(err, "error initializing cli")
 		}
 
+		gitRevision, err := cmd.Flags().GetString("git")
+		if err != nil {
+			return errors.Wrap(err, "could not read option --git")
+		}
+
 		var path string
 		if len(args) == 1 {
+			if gitRevision != "" {
+				cmd.SilenceUsage = false
```

Comment:
I think you need to unsilence in this case too:

```
./dist/epinio-linux-amd64 push --git main https://github.com/epinio/example-12factor.git
```
This is missing the app name but it prints no error, just exits.

Reply:
The code here should be unsilencing things. Double Negation: silence = false => do not silence = unsilence.
Will investigate.

Reply:
Hm. removing the statement makes no difference. No output ... disabling the `silence = true` at the top ... no difference ... I am beginning to hate cobra error handling.

Reply:
:bulb: I am using errors.Wrap, except, the `err` is likely nil :scream:

Reply:
Fixed with commit b526fc85e7baa8906a9444a9b828f1d36ff67625